File: composeApp/src/androidMain/kotlin/com/ivangarzab/kluvs/ui/me/MeScreenSkeleton.kt
package com.ivangarzab.kluvs.ui.me

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.HorizontalDivider
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Shape
import androidx.compose.ui.tooling.preview.PreviewLightDark
import androidx.compose.ui.unit.dp
import com.ivangarzab.kluvs.designsystem.components.loading.rememberShimmerColor
import com.ivangarzab.kluvs.designsystem.theme.KluvsTheme

/**
 * Full-screen placeholder shown while the profile loads — mirrors [MeScreenContent]'s shape
 * (profile row, stats row, up-next card, shelf rows) so nothing jumps around once the real
 * content swaps in.
 */
@Composable
fun MeScreenSkeleton(modifier: Modifier = Modifier) {
    val shimmerColor = rememberShimmerColor()

    fun Modifier.bone(shape: Shape = RoundedCornerShape(4.dp)) =
        this.clip(shape).background(shimmerColor)

    Column(
        modifier = modifier
            .background(color = KluvsTheme.colors.background)
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Column(modifier = Modifier.size(60.dp).bone(CircleShape)) {}
            Spacer(Modifier.padding(8.dp))
            Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Column(modifier = Modifier.width(120.dp).height(20.dp).bone()) {}
                Column(modifier = Modifier.width(80.dp).height(14.dp).bone()) {}
            }
        }

        HorizontalDivider(color = KluvsTheme.colors.divider)

        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceEvenly
        ) {
            repeat(3) {
                Column(
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(6.dp)
                ) {
                    Column(modifier = Modifier.width(32.dp).height(20.dp).bone()) {}
                    Column(modifier = Modifier.width(48.dp).height(12.dp).bone()) {}
                }
            }
        }

        HorizontalDivider(color = KluvsTheme.colors.divider)

        Column(
            modifier = Modifier.fillMaxWidth(),
            verticalArrangement = Arrangement.spacedBy(8.dp)
        ) {
            Column(modifier = Modifier.width(80.dp).height(12.dp).bone()) {}
            Column(modifier = Modifier.fillMaxWidth(0.7f).height(24.dp).bone()) {}
        }

        HorizontalDivider(color = KluvsTheme.colors.divider)

        Column(
            modifier = Modifier.fillMaxWidth(),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            Column(modifier = Modifier.width(100.dp).height(12.dp).bone()) {}
            repeat(2) {
                Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                    Column(
                        modifier = Modifier
                            .width(64.dp)
                            .height(96.dp)
                            .bone(RoundedCornerShape(4.dp))
                    ) {}
                    Column(
                        modifier = Modifier.weight(1f).padding(top = 4.dp),
                        verticalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        Column(modifier = Modifier.fillMaxWidth(0.6f).height(16.dp).bone()) {}
                        Column(modifier = Modifier.fillMaxWidth(0.4f).height(12.dp).bone()) {}
                    }
                }
            }
        }
    }
}

@PreviewLightDark
@Composable
private fun Preview_MeScreenSkeleton() = KluvsTheme {
    MeScreenSkeleton()
}

File: iosApp/iosApp/Views/Me/MeScreenSkeleton.swift
import SwiftUI
import DesignSystem

/// Full-screen placeholder shown while the profile loads — mirrors MeView's shape (profile
/// row, stats row, up-next card, shelf rows) so nothing jumps around once the real content
/// swaps in. Mirrors Android's `MeScreenSkeleton`.
struct MeScreenSkeleton: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 16) {
            HStack(alignment: .center) {
                Bone(Circle()).frame(width: 60, height: 60)
                VStack(alignment: .leading, spacing: 8) {
                    Bone().frame(width: 120, height: 20)
                    Bone().frame(width: 80, height: 14)
                }
                .padding(.leading, 8)
            }

            Divider()

            HStack {
                ForEach(0..<3, id: \.self) { _ in
                    VStack(spacing: 6) {
                        Bone().frame(width: 32, height: 20)
                        Bone().frame(width: 48, height: 12)
                    }
                    .frame(maxWidth: .infinity)
                }
            }

            Divider()

            VStack(alignment: .leading, spacing: 8) {
                Bone().frame(width: 80, height: 12)
                Bone().frame(width: 240, height: 24)
            }

            Divider()

            VStack(alignment: .leading, spacing: 12) {
                Bone().frame(width: 100, height: 12)
                ForEach(0..<2, id: \.self) { _ in
                    HStack(spacing: 12) {
                        Bone().frame(width: 64, height: 96)
                        VStack(alignment: .leading, spacing: 8) {
                            Bone().frame(width: 160, height: 16)
                            Bone().frame(width: 100, height: 12)
                        }
                        .padding(.top, 4)
                    }
                }
            }
        }
        .padding(16)
        .background(KluvsTheme.colors.background)
    }
}

#Preview {
    MeScreenSkeleton()
}
